Soleras3/react-challenge-project-jan-2023
# Update Dependencies

## Changes
1. Update outdated dependencies.
2. Fix breaking changes.
3. Move `react-scripts` to dev dependencies.

Note: the merge and revert from dev is because I accidentally re-used this branch. I chose to repurpose this branch for this PR, and made a new branch off a specific commit for this branch's original purpose. See: https://github.com/Soleras3/react-challenge-project-jan-2023/pull/1#issuecomment-1474468999

## Purpose
Projects must keep their libraries up to date regularly. Security vulnerabilities get fixed regularly, and small breaking changes become big headaches the longer a dependency goes without update.

See: https://github.com/Shift3/react-challenge-project-jan-2023/issues/7

## Approach
1. Update each dependency one-by-one and test each update for breaking changes.
2. Fix security concerns with `npm audit` and `npm audit fix --force`. (Most were incidentally fixed in #8.)
3. `react-scripts` specifically report false-positives that they will not update. Moving it to a dev dependency lets us audit via `npm audit --omit:dev`. See Learning section below.

Previous PRs dealing with routes and/or navigation will break with this update.

## Testing Steps
1. Run `npm install`
2. Start the app. Observe that it renders and actions work as expected.

## Learning
Standard procedure. `react-scripts` audits could not be fixed-- researching that, I found that FB devs consider dev-only dependency vulnerabilities to be almost always false-positives.

See this official `react-scripts` [recommendation](https://github.com/facebook/create-react-app/issues/11174).
See also [a rather caustic take](https://overreacted.io/npm-audit-broken-by-design/) from FB engineer Dan Abramov.

Closes https://github.com/Shift3/react-challenge-project-jan-2023/issues/7
